xformers/components/swiglu/cuda/silu_bw_fused.cu

```diff
@@ -0,0 +1,98 @@
+#include <ATen/Tensor.h>
+#include <ATen/Dispatch.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/ScalarOps.h>
+#include <c10/cuda/CUDAGuard.h>
+#include <torch/library.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/ReduceOps.h>
+#include <ATen/native/Resize.h>
+#include <ATen/native/cuda/Loops.cuh>
+
+
+namespace {
+/*
+Computes the following:
+
+def silu_bw_fused(x1, x2, dx4):
+    x3 = F.silu(x1)
+    dx3 = dx4 * x2
+    dx2 = dx4 * x3
+    x4 = x2 * x3  # checkpointing
+    # silu bw
+    sigm = 1 / (1 + torch.exp(-x1.float()))
+    dx1 = (dx3.float() * sigm * (1 + x1.float() * (1 - sigm))).to(x1.dtype)
+    return dx1, dx2, x4
+*/
+
+template <typename T>
+struct KernelTraits {
+  using AccumulationElement = T;
+};
+
+template <>
+struct KernelTraits<at::Half> {
+  using AccumulationElement = float;
+};
+
+template <>
+struct KernelTraits<at::BFloat16> {
+  using AccumulationElement = float;
+};
```

Comment:
nit: you can use `AccumulateType` (or `acc_type` for short) [from PyTorch here instead](https://github.com/pytorch/pytorch/blob/36380897553063c7b433f738671ff23c5ad58ced/aten/src/ATen/AccumulateType.h)